web/html/src/components/input/InputBase.tsx

```diff
@@ -129,7 +129,7 @@ export class InputBase extends React.Component<Props, State> {
           }
           return filtered;
         }, {});
-        this.validate(values);
+        this.validate(values as (T extends {} ? T : never));
```

Comment:
The problem here is a bit involved and stems from the one-thing-does-all nature of `InputBase`. `InputBase` is a generic with `T = string` which annotates that usually the value an input gives out is a string. However, with the array names thing Cedric added, it also supports objects and pretty much anything else.  
This code path can only be hit in the latter case (`if(this.props.name instanceof Array) ...`), but the type engine won't know what that implies.  

All that said, this is all hacky, but I don't have a better way to annotate what's going on without refactoring the code to be more sane — I'm trying to avoid changing the logic right now because I don't want to introduce regressions.  
I'll try to think of a comment that will explain all of this.

Reply:
For now, I've sort of hidden the complexity behind making `this.validate()` a generic. The type complexity is similar, but in this case it's contained in a single place.